File: REST/Models/Leg.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace CosmosOdyssey.REST.Models
{
    public class Leg
    {
        public Guid Id { get; set; }
        public RouteInfo RouteInfo { get; set; }
        public Guid RouteInfoId { get; set; } //FK
        public ICollection<Provider> Providers { get; set; } = new List<Provider>(); //Üks-mitmele seos Offeritega // vb peaks kustutama
        public PriceList PriceList { get; set; }
        public Guid PriceListId { get; set; } // FK

    }
}

File: Frontend/cosmosodyssey/types/leg.ts
import type {RouteInfo} from './routeInfo';
import type {Provider} from './provider';
import type {PriceList} from './priceList';
import type {Guid} from './guid';


export type Leg = {
    id: Guid;
    routeInfo: RouteInfo;
    routeInfoId: Guid;
    providers: Provider[];
    priceList: PriceList;
    priceListId: Guid;
}